photutils/psf/models.py

```diff
@@ -12,8 +12,274 @@
 from ..utils import mask_to_mirrored_num
 from ..extern.nddata_compat import extract_array
 
+__all__ = ['Discrete2DModel', 'IntegratedGaussianPRF', 'PRFAdapter',
+           'prepare_psf_model']
 
-__all__ = ['IntegratedGaussianPRF', 'PRFAdapter', 'prepare_psf_model']
+
+class Discrete2DModel(Fittable2DModel):
+    """
+    A discrete fittable 2D model of an image.
+
+    This class stores a discrete 2D image and computes the values at arbitrary
+    locations (including at intra-pixel, fractional positions) within this
+    image using spline interpolation provided by
+    :py:class:`~scipy.interpolate.RectBivariateSpline`. Even though this
+    particular spline interpolator does not support weighted smoothing,
+    :py:class:`Discrete2DModel` can be used to store image weights
+    so that these can be passed to the fitter.
+
+    The fittable model provided by this class has three model parameters:
+    total flux of the underlying image, and two shifts along each axis of the
```

Comment:
I found the phrasing "total flux of the underlying image" to be a bit confusing.  In my mind "the underlying image" is the thing that you've *fit*, not the output of the model.

I would instead suggest the parameters be described how they are meant to be *used*  I.e. "the total flux in the modeled object, and the center of the object".  I realize that *internally* that corresponds to shifts along the axis, but that's not how you view the *model* itself, which is what 99% of users are going to be concerned with.